### PR TITLE
rewrites to optionals

### PR DESCRIPTION
**Pull request by ingmargoudt**

### CODE REVIEW

**Comment by LevelX2** [comment]
I guess the following problem is caused by this pull request.

INFO  2017-03-20 22:58:15,886 Logged-in as Bandit to MAGE server at localhost:17171                                      =>[SwingWorker-pool-4-thread-2] SessionImpl$4.run 
java.lang.InterruptedException: sleep interrupted
	at java.lang.Thread.sleep(Native Method)
	at mage.client.dialog.UpdateSeatsTask.lambda$doInBackground$0(TableWaitingDialog.java:426)
	at java.util.Optional.ifPresent(Optional.java:159)
	at mage.client.dialog.UpdateSeatsTask.doInBackground(TableWaitingDialog.java:422)
	at mage.client.dialog.UpdateSeatsTask.doInBackground(TableWaitingDialog.java:402)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
INFO  2017-03-20 22:58:22,215 Game c935b1f3-d6b8-4206-aad9-b3273d0fc719 started for player 5230f413-9168-4cbe-aa90-1dcd39203d98 =>[AWT-EventQueue-0] CallbackClientImpl.gameStarted

**Comment by ingmargoudt** [comment]
I checked the line and compared it with previous versions. 

previous:
 protected Void doInBackground() throws Exception {
        while (!isCancelled()) {
            this.publish(SessionHandler.getTable(roomId, tableId));
            Thread.sleep(1000);
        }
        return null;
    }


now: 

protected Void doInBackground() throws Exception {
        while (!isCancelled()) {
            SessionHandler.getTable(roomId, tableId).ifPresent(tableView -> {

                this.publish(tableView);
                try {
                    Thread.sleep(1000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            });
        }
        return null;
    }

So basically, now that  there is a lambda, the throws exception from the method signature can not be reached, instead the InterruptException (which has always been there) is caught and the stacktrace is printed out
